core/src/main/java/org/infinispan/persistence/spi/NonBlockingStore.java

```diff
@@ -580,25 +586,23 @@
     *    </tr>
     *    <tr>
     *       <td valign="top">{@link Characteristic#SEGMENTABLE}</td>
-    *       <td valign="top">When this is not set or segmentation is disabled in the
+    *       <td valign="top">When the store does not have this characteristic or segmentation is disabled in the
     *       {@link StoreConfiguration#segmented() configuration},
-    *       the {@code segment} parameter may be ignored.</td>
+    *       the {@code segment} parameter is always {@code IntSets.immutableRangeSet(numSegments)}.</td>
     *    </tr>
     * </table>
     * <p>
     * If a problem is encountered, it is recommended to wrap any created/caught Throwable in a
     * {@link PersistenceException} and the stage be completed exceptionally.
     * <p>
     * @implSpec
-    * A default implementation is provided that does the following:
-    * <pre>{@code
-    * return size(segments);}
-    * </pre>
+    * A default implementation is provided that always returns {@code -1}.
     * @param segments the segments for which the entries are counted.
-    * @return a stage that, when complete, contains the count of how many entries are present for the given segments.
+    * @return a stage that, when complete, contains the approximate count of the entries in the given segments,
+    * or {@code -1L} if an approximate count cannot be provided.
     */
    default CompletionStage<Long> approximateSize(IntSet segments) {
-      return size(segments);
+      return CompletableFuture.completedFuture(-1L);
```

Comment:
```suggestion
      return SIZE_UNAVAILABLE_FUTURE;
```

Reply:
I didn't try to optimize the default implementations thinking stores should override the default, and even if they didn't, one allocation wouldn't matter for a store. I used a constant in PersistenceManagerImpl only because that method can be invoked when there is no store.

Reply:
I think it might be useful to add the -1 future to CompletableFutures, but I am fine either way.